### PR TITLE
Feat/57 perms and roles endpoints

### LINKED ISSUE #57: [FEAT] API Endpoints voor Rollen en Permissies
### Korte beschrijving:
Zorgen dat beheerders via de API rollen kunnen bekijken, aanmaken en aanpassen. Op dit moment staan de rollen wel in de database, maar kunnen we er van buitenaf nog niets mee.

### Lijst van dingen die gedaan moeten zijn
- Werkende API endpoints voor rollen en permissies.

### PR DESCRIPTION
**Pull request by ARm8-2**
### Beschrijving
Backend api endpoints toegevoegd om rollen te beheren, later nodig voor het beheren van users.

Closes #57 

- [x] Goede testen toegevoegd.
- [ ] Auteur wil zelf mergen.

### CODE REVIEW

File: backend/src/services/auth/role.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    if role.permissions:
+        valid_permissions = set(Permissions.all())
+        invalid = [p for p in role.permissions if p not in valid_permissions]
+        if invalid:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=f"Invalid permissions: {', '.join(invalid)}",
+            )
+        perms = db.query(Permission).filter(Permission.name.in_(role.permissions)).all()
+        if len(perms) != len(role.permissions):
```
Comment (PurpleCosmic) [inline backend/src/services/auth/role.py]: Als er voor een of andere reden duplicate permissions meegegeven worden geeft dit dezelfde error als het geval waarin een permission niet in de databank zit. Dit kan mogelijks voor onduidelijkheden zorgen, ik stel voor hiervoor een aparte error te geven of de lijst van perms te deduplicaten.

File: backend/src/services/auth/role.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        invalid = [p for p in update.permissions if p not in valid_permissions]
+        if invalid:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=f"Invalid permissions: {', '.join(invalid)}",
+            )
+        perms = (
+            db.query(Permission).filter(Permission.name.in_(update.permissions)).all()
+        )
+        if len(perms) != len(update.permissions):
```
Comment (PurpleCosmic) [inline backend/src/services/auth/role.py]: idem comment op line R33
(duplicate permissions)

File: backend/tests/integration/test_roles_permissions_endpoints.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        "/api/v1/auth/roles/",
+        json={"name": "testrole", "permissions": [Permissions.USERS_READ]},
+        headers=headers,
+    )
+    assert response.status_code == 201
+    role_id = response.json()["id"]
+
+    response = client.get(f"/api/v1/auth/roles/{role_id}", headers=headers)
+    assert response.status_code == 200
+    assert response.json()["name"] == "testrole"
```
Comment (PurpleCosmic) [inline backend/tests/integration/test_roles_permissions_endpoints.py:99]: Zou het niet nuttig zijn ook te testen of alle permissions wel degelijk in de response permissions aanwezig zijn na het aanmaken van een rol?

**Review by PurpleCosmic — CHANGES_REQUESTED** [review]
Enige opmerking is een onbehandelde edge case maar voor de rest ziet alles er goed uit!

File: backend/tests/integration/test_roles_permissions_endpoints.py (hunk 3 above)
Comment (ARm8-2) [reply backend/tests/integration/test_roles_permissions_endpoints.py:99]: Jup, toegevoegd

File: backend/src/services/auth/role.py (hunk 1 above)
Comment (ARm8-2) [reply backend/src/services/auth/role.py]: Mhmm, permissies worden nu eerst in een set geplaatst om ervoor te zorgen dat ze uniek zijn.

File: backend/src/services/auth/role.py (hunk 2 above)
Comment (ARm8-2) [reply backend/src/services/auth/role.py]: Zelfde als vorige comment ook

**Review by PurpleCosmic — APPROVED** [review]

**Review by Nathan-verm — APPROVED** [review]
Hoe ik het altijd deed voor de exceptions:  raise ValueError in de service file en dan die ValueError opvangen in de router file en dan pas een HTTP exception te gooien. Maar in principe maakt het niet zoveel uit.
